packages/webapp/src/containers/Finances/index.jsx

```diff
@@ -63,19 +57,26 @@ const Finances = ({ history }) => {
   const transactions = useTransactions({ dateFilter, expenseTypeFilter, revenueTypeFilter });
   const currencySymbol = useCurrencySymbol();
   const overlayRef = useRef(null);
+  const isFetchingData = useSelector(isFetchingDataSelector);
+  const [isLoading, setIsLoading] = useState(true);
 
   const dispatch = useDispatch();
 
   useEffect(() => {
-    dispatch(getFarmExpenseType());
-    dispatch(getRevenueTypes());
-    dispatch(getSales());
-    dispatch(getExpense());
-    dispatch(getManagementPlansAndTasks());
-    dispatch(getCropVarieties());
-    dispatch(setSelectedExpenseTypes([]));
-    dispatch(resetTransactionsFilter());
-    dispatch(setDateRange({ option: dateRangeOptions.YEAR_TO_DATE }));
+    dispatch(fetchAllData());
+  }, [fetchAllData]);
+
+  useEffect(() => {
+    if (!isFetchingData) {
+      setIsLoading(false);
+    }
+  }, [isFetchingData]);
+
+  useEffect(() => {
+    return () => {
+      setIsLoading(true);
```

Comment:
I don't understand why the state needs to be updated when unmounting...
@kathyavini do you have any ideas?

Reply:
Hmmm I don't know! But with that cleanup you do get a loader when you go from Finances > Add Sale and then back button, don't you? 🤔 I don't like that! 😆

Reply:
You see the spinner when going back, after adding revenue/expense... whenever you go to the finance page. I don't like that either 😅 

I don't think unmounted component keeps its state, and I think it's always started with the initial value `true` passed to `useState`. I don't see any difference even if I remove `setIsLoading(true);`...

I'm reviewing your change now...

Reply:
Oh yes sorry I used state too loosely there! The component state `isLoading` always resets to true. I just added the reducer bit to the other saga because that one actually does seem to control whether the spinner appears.